lib/jnpr/junos/decorators.py

```diff
@@ -119,14 +119,20 @@ def wrapper(*args, **kwargs):
                             if isinstance(ignore_warn, (str, unicode)):
                                 if not re.search(ignore_warn, err['message'],
                                                  re.I):
+                                    # Message did not match.
                                     raise ex
                             elif isinstance(ignore_warn, list):
                                 for warn_msg in ignore_warn:
-                                    if not re.search(warn_msg, err['message'],
-                                                     re.I):
-                                        raise ex
+                                    if re.search(warn_msg, err['message'],
+                                                 re.I):
+                                        # Warning matches. Break skips else.
+                                        break
```

Comment:
say we have errors as
```
[{'severity': 'warning', 'edit_path': None, 'message': 'Advertisement-interval is less than four', 'bad_element':'l2cpd', 'source': ‘l2cpd’},
 {'severity': 'warning', 'edit_path': None, 'message': 'A system reboot is mandatory', 'bad_element': 'chassis', 'source': ‘chassisd’}]
```
user pass
ignore_warning = ['A system reboot is mandatory', 'xxxx', 'yyyy']

here the device returned 2 warnings, one matches what being passed in the list, other "'Advertisement-interval is less than four'" is not being match. With a break statment, we are just ignoring 2nd warning. does our ignore_warning means that even if any of the warnings from Device matches any from the list give by user, We are good?